src/components/shop-app.js

```diff
@@ -0,0 +1,507 @@
+    if (page === 'list') {
+      scrollTop = this._listScrollTop;
+    }
+    // Use `Polymer.AppLayout.scroll` with `behavior: 'silent'` to disable header scroll
+    // effects during the scroll.
+    scroll({ top: scrollTop, behavior: 'silent' });
+
+    switch (page) {
+      case 'home':
+        store.dispatch(updateMeta({ title: 'Home' }));
+        break;
+      case 'list':
+        await import('../components/shop-list.js');
+        if (this._category) {
+          store.dispatch(updateMeta({
+            title: this._category.title,
+            image: document.baseURI + this._category.image
+          }));
+        }
+        break;
+      case 'detail':
+        await import('../components/shop-detail.js');
+        if (this._item) {
+          store.dispatch(updateMeta({
+            title: this._item.title,
+            description: this._item.description.substring(0, 100),
+            image: document.baseURI + this._item.image
+          }));
+        }
+        break;
+      case 'cart':
+        await import('../components/shop-cart.js');
+        store.dispatch(updateMeta({ title: 'Cart' }));
+        break;
+      case 'checkout':
+        await import('../components/shop-checkout.js');
+        store.dispatch(updateMeta({ title: 'Checkout' }));
+        break;
+      default:
+        store.dispatch(updateMeta({ title: '404' }));
+    }
+
+    this._ensureLazyLoaded();
+    if (oldPage) {
+      // The size of the header depends on the page (e.g. on some pages the tabs
+      // do not appear), so reset the header's layout when switching pages.
+      timeOut.run(() => {
+        const header = this.shadowRoot.querySelector('#header');
+        header.resetLayout();
+      }, 1);
+    }
+  }
+
+  _ensureLazyLoaded() {
+    // load lazy resources after render and set `loadComplete` when done.
+    if (!this.loadComplete) {
+      afterNextRender(this, () => {
+        import('./lazy-resources.js').then(() => {
+          // Register service worker if supported.
+          if ('serviceWorker' in navigator) {
+            navigator.serviceWorker.register('service-worker.js', {scope: '/'});
+          }
+          this.loadComplete = true;
+          // The size of the header depends on the page and whether tabs have loaded,
+          // so reset the header's layout after load completes.
+          timeOut.run(() => {
+            const header = this.shadowRoot.querySelector('#header');
+            header.resetLayout();
+          }, 1);
+        });
+      });
+    }
+  }
+
+  _updateLocation() {
+    store.dispatch(updateLocation(window.decodeURIComponent(window.location.pathname)));
+
+    // Close the drawer - in case the *route* change came from a link in the drawer.
+    this.drawerOpened = false;
+  }
+
+  _offlineChanged(offline, oldOffline) {
+    // Show the snackbar if the user is offline when starting a new session
+    // or if the network status changed.
+    if (offline || (!offline && oldOffline === true)) {
```

Comment:
I believe @kevinpschaaf had a similar construction in the real estate app and made this explicitly part of the store to avoid having to toggle it ad hoc like this.